crates/solver/src/solver/naive_solver.rs

```diff
@@ -445,25 +445,19 @@ mod tests {
             }),
         ];
 
-        let amm_handler = CapturingSettlementHandler::arc();
         let tokens = TokenPair::new(usdc, crv).unwrap();
         let liquidity = hashmap! {
             tokens => ConstantProductOrder {
                 address: addr!("210a97ba874a8e279c95b350ae8ba143a143c159"),
                 tokens,
                 reserves: (32275540, 33308141034569852391),
                 fee: Ratio::new(3, 1000),
-                settlement_handling: amm_handler.clone(),
+                settlement_handling: CapturingSettlementHandler::arc(),
             },
         };
 
-        settle(SlippageContext::default(), orders, liquidity);
-        let TokenAmount {
-            token: out_token,
-            amount: out_amount,
-        } = amm_handler.calls.lock().unwrap()[0].output;
-        assert_eq!(out_token, usdc);
-        assert!(out_amount < U256::from(32275540_u128));
+        let settlements = settle(SlippageContext::default(), orders, liquidity);
+        assert!(settlements.is_empty());
```

Comment:
This test's assertion was useless. It was capturing the last swap amounts, but there was no settlement so that data invalid anyway.

It turns out that this settling these orders is actually impossible, so assert that instead.